builder/dockerfile/evaluator.go

```diff
@@ -225,6 +160,8 @@ func (s *dispatchState) beginStage(stageName string, image builder.Image) {
 	}
 	s.baseImage = image
 	s.setDefaultPath()
+	s.runConfig.OpenStdin = false
+	s.runConfig.StdinOnce = false
```

Comment:
Ah, nice. I hadn't realized these were previously mixed up in the onBuild logic. This is a much better place for them.